File: native/swift/swift-export-standalone-integration-tests/simple/testData/generation/protocols/golden_result/funinterface/funinterface.kt
@file:kotlin.Suppress("DEPRECATION_ERROR")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface.FunctorClass::class, "22ExportedKotlinPackages12funinterfaceO12funinterfaceE12FunctorClassC")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface.FunctionalInterface::class, "_FunctionalInterface")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation::class, "_XMLFunctionalInterfaceWithLeadingAbbreviation")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface._123FunctionalInterfaceWithLeadingNumbers::class, "__123FunctionalInterfaceWithLeadingNumbers")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation::class, "__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface._FunctionalInterfaceWithLeadingUnderscore::class, "__FunctionalInterfaceWithLeadingUnderscore")
@file:kotlin.native.internal.objc.BindClassToObjCName(funinterface.functionalInterfaceWithAlreadyLowercaseLeading::class, "_functionalInterfaceWithAlreadyLowercaseLeading")

import kotlin.native.internal.ExportedBridge
import kotlinx.cinterop.*
import kotlinx.cinterop.internal.convertBlockPtrToKotlinFunction

@ExportedBridge("funinterface_FunctionalInterface__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface_FunctionalInterface__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface.FunctionalInterface(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface_FunctionalInterface_invoke")
public fun funinterface_FunctionalInterface_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface.FunctionalInterface
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface_FunctorClass_init_allocate")
public fun funinterface_FunctorClass_init_allocate(): kotlin.native.internal.NativePtr {
    val _result = kotlin.native.internal.createUninitializedInstance<funinterface.FunctorClass>()
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface_FunctorClass_init_initialize__TypesOfArguments__Swift_UnsafeMutableRawPointer__")
public fun funinterface_FunctorClass_init_initialize__TypesOfArguments__Swift_UnsafeMutableRawPointer__(__kt: kotlin.native.internal.NativePtr): Unit {
    val ____kt = kotlin.native.internal.ref.dereferenceExternalRCRef(__kt)!!
    kotlin.native.internal.initInstance(____kt, funinterface.FunctorClass())
}

@ExportedBridge("funinterface_FunctorClass_invoke")
public fun funinterface_FunctorClass_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface.FunctorClass
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation_invoke")
public fun funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface__123FunctionalInterfaceWithLeadingNumbers__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface__123FunctionalInterfaceWithLeadingNumbers__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface._123FunctionalInterfaceWithLeadingNumbers(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface__123FunctionalInterfaceWithLeadingNumbers_invoke")
public fun funinterface__123FunctionalInterfaceWithLeadingNumbers_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface._123FunctionalInterfaceWithLeadingNumbers
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation_invoke")
public fun funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface__FunctionalInterfaceWithLeadingUnderscore__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface__FunctionalInterfaceWithLeadingUnderscore__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface._FunctionalInterfaceWithLeadingUnderscore(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface__FunctionalInterfaceWithLeadingUnderscore_invoke")
public fun funinterface__FunctionalInterfaceWithLeadingUnderscore_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface._FunctionalInterfaceWithLeadingUnderscore
    val _result = __self.invoke()
    return _result
}

@ExportedBridge("funinterface_functionalInterfaceWithAlreadyLowercaseLeading__TypesOfArguments__U2829202D_U20Swift_Int32__")
public fun funinterface_functionalInterfaceWithAlreadyLowercaseLeading__TypesOfArguments__U2829202D_U20Swift_Int32__(function: kotlin.native.internal.NativePtr): kotlin.native.internal.NativePtr {
    val __function = run {
        val kotlinFun = convertBlockPtrToKotlinFunction<()->Int>(function);
        { kotlinFun() }
    }
    val _result = funinterface.functionalInterfaceWithAlreadyLowercaseLeading(__function)
    return kotlin.native.internal.ref.createRetainedExternalRCRef(_result)
}

@ExportedBridge("funinterface_functionalInterfaceWithAlreadyLowercaseLeading_invoke")
public fun funinterface_functionalInterfaceWithAlreadyLowercaseLeading_invoke(self: kotlin.native.internal.NativePtr): Int {
    val __self = kotlin.native.internal.ref.dereferenceExternalRCRef(self) as funinterface.functionalInterfaceWithAlreadyLowercaseLeading
    val _result = __self.invoke()
    return _result
}

File: native/swift/swift-export-standalone-integration-tests/simple/testData/generation/protocols/golden_result/funinterface/funinterface.swift
@_exported import ExportedKotlinPackages
@_implementationOnly import KotlinBridges_funinterface
import KotlinRuntime
import KotlinRuntimeSupport

extension ExportedKotlinPackages.funinterface.FunctionalInterface where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface_FunctionalInterface_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface.FunctionalInterface {
}
extension ExportedKotlinPackages.funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation {
}
extension ExportedKotlinPackages.funinterface._123FunctionalInterfaceWithLeadingNumbers where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface__123FunctionalInterfaceWithLeadingNumbers_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface._123FunctionalInterfaceWithLeadingNumbers {
}
extension ExportedKotlinPackages.funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation {
}
extension ExportedKotlinPackages.funinterface._FunctionalInterfaceWithLeadingUnderscore where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface__FunctionalInterfaceWithLeadingUnderscore_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface._FunctionalInterfaceWithLeadingUnderscore {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface.FunctionalInterface where Wrapped : ExportedKotlinPackages.funinterface._FunctionalInterface {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface._FunctionalInterfaceWithLeadingUnderscore where Wrapped : ExportedKotlinPackages.funinterface.__FunctionalInterfaceWithLeadingUnderscore {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface._123FunctionalInterfaceWithLeadingNumbers where Wrapped : ExportedKotlinPackages.funinterface.__123FunctionalInterfaceWithLeadingNumbers {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation where Wrapped : ExportedKotlinPackages.funinterface._XMLFunctionalInterfaceWithLeadingAbbreviation {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation where Wrapped : ExportedKotlinPackages.funinterface.__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation {
}
extension KotlinRuntimeSupport._KotlinExistential: ExportedKotlinPackages.funinterface.functionalInterfaceWithAlreadyLowercaseLeading where Wrapped : ExportedKotlinPackages.funinterface._functionalInterfaceWithAlreadyLowercaseLeading {
}
extension ExportedKotlinPackages.funinterface.functionalInterfaceWithAlreadyLowercaseLeading where Self : KotlinRuntimeSupport._KotlinBridgeable {
    public func callAsFunction() -> Swift.Int32 {
        return funinterface_functionalInterfaceWithAlreadyLowercaseLeading_invoke(self.__externalRCRef())
    }
}
extension ExportedKotlinPackages.funinterface.functionalInterfaceWithAlreadyLowercaseLeading {
}
extension ExportedKotlinPackages.funinterface {
    public protocol FunctionalInterface: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    public protocol XMLFunctionalInterfaceWithLeadingAbbreviation: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    public protocol _123FunctionalInterfaceWithLeadingNumbers: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    public protocol _123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    @objc(_FunctionalInterface)
    package protocol _FunctionalInterface {
    }
    public protocol _FunctionalInterfaceWithLeadingUnderscore: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    @objc(_XMLFunctionalInterfaceWithLeadingAbbreviation)
    package protocol _XMLFunctionalInterfaceWithLeadingAbbreviation {
    }
    @objc(__123FunctionalInterfaceWithLeadingNumbers)
    package protocol __123FunctionalInterfaceWithLeadingNumbers {
    }
    @objc(__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation)
    package protocol __123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation {
    }
    @objc(__FunctionalInterfaceWithLeadingUnderscore)
    package protocol __FunctionalInterfaceWithLeadingUnderscore {
    }
    @objc(_functionalInterfaceWithAlreadyLowercaseLeading)
    package protocol _functionalInterfaceWithAlreadyLowercaseLeading {
    }
    public protocol functionalInterfaceWithAlreadyLowercaseLeading: KotlinRuntime.KotlinBase {
        func callAsFunction() -> Swift.Int32
    }
    public final class FunctorClass: KotlinRuntime.KotlinBase, ExportedKotlinPackages.funinterface.FunctionalInterface, ExportedKotlinPackages.funinterface._FunctionalInterface {
        public init() {
            if Self.self != ExportedKotlinPackages.funinterface.FunctorClass.self { fatalError("Inheritance from exported Kotlin classes is not supported yet: \(String(reflecting: Self.self)) inherits from ExportedKotlinPackages.funinterface.FunctorClass ") }
            let __kt = funinterface_FunctorClass_init_allocate()
            super.init(__externalRCRefUnsafe: __kt, options: .asBoundBridge)
            funinterface_FunctorClass_init_initialize__TypesOfArguments__Swift_UnsafeMutableRawPointer__(__kt)
        }
        package override init(
            __externalRCRefUnsafe: Swift.UnsafeMutableRawPointer?,
            options: KotlinRuntime.KotlinBaseConstructionOptions
        ) {
            super.init(__externalRCRefUnsafe: __externalRCRefUnsafe, options: options)
        }
        public func callAsFunction() -> Swift.Int32 {
            return funinterface_FunctorClass_invoke(self.__externalRCRef())
        }
    }
    public static func _123functionalInterfaceWithLeadingNumbers(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface._123FunctionalInterfaceWithLeadingNumbers {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface__123FunctionalInterfaceWithLeadingNumbers__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface._123FunctionalInterfaceWithLeadingNumbers
    }
    public static func _123xmlFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface__123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface._123XMLFunctionalInterfaceWithLeadingUnderscoreNumbersAndAbbreviation
    }
    public static func _functionalInterfaceWithLeadingUnderscore(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface._FunctionalInterfaceWithLeadingUnderscore {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface__FunctionalInterfaceWithLeadingUnderscore__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface._FunctionalInterfaceWithLeadingUnderscore
    }
    public static func functionalInterface(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface.FunctionalInterface {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface_FunctionalInterface__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface.FunctionalInterface
    }
    public static func functionalInterfaceWithAlreadyLowercaseLeadingFromFunction(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface.functionalInterfaceWithAlreadyLowercaseLeading {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface_functionalInterfaceWithAlreadyLowercaseLeading__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface.functionalInterfaceWithAlreadyLowercaseLeading
    }
    public static func xmlFunctionalInterfaceWithLeadingAbbreviation(
        function: @escaping () -> Swift.Int32
    ) -> any ExportedKotlinPackages.funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation {
        return KotlinRuntime.KotlinBase.__createProtocolWrapper(externalRCRef: funinterface_XMLFunctionalInterfaceWithLeadingAbbreviation__TypesOfArguments__U2829202D_U20Swift_Int32__({
            let originalBlock = function
            return { return originalBlock() }
        }())) as! any ExportedKotlinPackages.funinterface.XMLFunctionalInterfaceWithLeadingAbbreviation
    }
}
